### ISSUE TITLE
ToDecimal(LossOfPrecisionHandling.Truncate) performs rounding and not truncation. 

### ISSUE THREAD

**Issue by Rishabh-V** [googleapis/google-cloud-dotnet#8285]
This was discovered and pointed out by @jskeet  while reviewing the [code for PgNumeric](https://github.com/googleapis/google-cloud-dotnet/pull/8261) but applies to `SpannerNumeric`, `BigQueryNumeric`, `BigQueryBigNumeric` as well.

Issue: `ToDecimal` API of `PgNumeric` called with the parameter `LossOfPrecisionHandling.Truncate`  performs rounding of numeric value rather than truncation as one would expect by passing the parameter as `LossOfPrecisionHandling.Truncate` 

E.g.
```
var numeric = PgNumeric.Parse("123456789012345678900.123456789");
var decimalValue = numeric.ToDecimal(LossOfPrecisionHandling.Truncate);
```
One would expect the `decimalValue` to be 123456789012345678900.12345678
However, actual value is rounded to 123456789012345678900.1234567*9*

The reason for this behavior is that the code of `ToDecimal()` first calls the `ToString()` method to convert numeric value to string and then parses the string into decimal by calling `decimal.Parse` which [performs rounding and not truncation](https://docs.microsoft.com/en-us/dotnet/api/system.decimal.parse?view=netstandard-2.1) 

The code fix needs to be made in PgNumeric, SpannerNumeric, BigQueryNumeric, BigQueryBigNumeric (and other numeric implementations if any)

**Comment by amanda-tarafa** [comment]
I'm grabbing this one. I have an alternative implementation that was supposed to improve performance and does so only slightly, but it does truncate.

I'll dust it off when I'm back.

**Comment by jskeet** [comment]
Okay - I was going to do it just by truncating the string before parsing... I suspect that would be somewhat simpler :)

**Comment by amanda-tarafa** [comment]
Well that's fine by me as well 😀

**Comment by jskeet** [comment]
@amanda-tarafa: Having tried to implement this, it's trickier than I'd expected and I think Numeric.cs needs a little more documentation, so I'll assign this to you for now, but let's discuss it together :)

**Comment by jskeet** [comment]
(One option would be to just rename the enum values for the next major version, in fact, and keep letting decimal do what it wants! I suspect very few people are actually using this directly.)

**Comment by jskeet** [comment]
Labelled this as release blocking in case we want to rename the enum value. We could potentially make it a non-breaking change my adding a new "name" for the same numeric value, and making the existing name obsolete.

**Comment by jskeet** [comment]
Note that we accept a `LossOfPrecisionHandling` in `FromDecimal` as well, so there are multiple paths to consider here.

**Comment by jskeet** [comment]
`FromDecimal` currently *does* truncate, whereas `ToDecimal` effectively applies banker's rounding.

Options:

- Just do nothing
- Provide a enum name that is pretty vague, but without changing behavior
- Change ToDecimal to *actually* truncate
- Change FromDecimal to round

**Comment by jskeet** [comment]
My current preference is actually to change ToDecimal to truncate. This requires a bit of string manipulation before parsing, basically. But I may well be missing something...

**Comment by amanda-tarafa** [comment]
+1 to change ToDecimal to truncate.

I still have my commit from trying to improve performance, and that one actually truncates (it might be improvable on other things though): https://github.com/googleapis/google-cloud-dotnet/compare/main...amanda-tarafa:bigquery-todecimal

**Comment by jskeet** [comment]
@amanda-tarafa: Any thoughts as to which approach you'd prefer us to take? I'm happy either way - in some ways, if we decide to *actually* truncate, there's less urgency as we don't need anything like a breaking change. (It'd be *nice* to get it into the new major version though.)

**Comment by amanda-tarafa** [comment]
Definetely have ToDecimal truncate.

I think it's best if we go with a ToString approach before GaxV4 and after we can try to improve performance if there's an issue for real. I'll try to push a PR that truncates directly on `ToString` and see how that looks.

**Comment by amanda-tarafa** [comment]
Fix for BigQuery has been merged in #8543.

Only Spanner pending, I'll do Spanner early next week at the latest.
